system-tests/ft-suite/src/test/kotlin/com/hotels/styx/proxy/CompressionSpec.kt

```diff
@@ -0,0 +1,165 @@
+/*
+  Copyright (C) 2013-2019 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.proxy
+
+import com.hotels.styx.api.HttpHeaderNames.HOST
+import com.hotels.styx.api.HttpRequest.get
+import com.hotels.styx.api.HttpResponseStatus.OK
+import com.hotels.styx.client.StyxHttpClient
+import com.hotels.styx.support.StyxServerProvider
+import com.hotels.styx.support.proxyHttpHostHeader
+import io.kotlintest.Spec
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.FeatureSpec
+import reactor.core.publisher.toMono
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.zip.GZIPInputStream
+
+class CompressionSpec : FeatureSpec() {
+
+    init {
+        feature("Content-type of the response is compressible") {
+            scenario("Compresses HTTP response if client requested gzip accept-encoding") {
```

Comment:
Is it worth testing that an unsupported accept-encoding scheme results in no compression?